pandas/tests/frame/indexing/test_categorical.py

```diff
@@ -117,7 +117,11 @@ def test_assigning_ops(self):
         tm.assert_frame_equal(df, exp_single_cats_value)
 
         #   - assign a single value not in the current categories set
-        with pytest.raises(ValueError):
+        msg = (
+            r"(:?Cannot setitem on a Categorical with a new category.*)"
+            r"|(:?Cannot set a Categorical with another, without identical categories)"
+        )
```

Comment:
Can this message be split up between the tests that follow it? It looks like the first part might only refer to some tests, while the second one only to others